falkordb_gemini_kg/steps/create_ontology_step.py

```diff
@@ -104,26 +97,42 @@ def _process_source(
 
         responses.append(self._call_model(chat_session, user_message))
 
-        logger.debug(f"Model response: {responses[response_idx].text}")
+        logger.debug(f"Model response: {responses[response_idx]}")
 
-        while (
-            responses[response_idx].candidates[0].finish_reason
-            == FinishReason.MAX_TOKENS
-        ):
+        while responses[response_idx].finish_reason == FinishReason.MAX_TOKENS:
             response_idx += 1
             responses.append(self._call_model(chat_session, "continue"))
 
-        if responses[response_idx].candidates[0].finish_reason != FinishReason.STOP:
+        if responses[response_idx].finish_reason != FinishReason.STOP:
             raise Exception(
-                f"Model stopped unexpectedly: {responses[response_idx].candidates[0].finish_reason}"
+                f"Model stopped unexpectedly: {responses[response_idx].finish_reason}"
             )
 
         combined_text = " ".join([r.text for r in responses])
 
         try:
-            new_ontology = Ontology.from_json(extract_json(combined_text))
+            data = json.loads(extract_json(combined_text))
+        except json.decoder.JSONDecodeError as e:
+            logger.debug(f"Error extracting JSON: {e}")
```

Comment:
How is it possible for the JSON to be invalid

Reply:
If the model did not respond with a valid json and we can't fix it. Can happen.